persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -247,33 +247,40 @@ public static void inferRelationDetails(Module entityModule) {
                             assocEntity.getFields().stream().filter(assocfield -> assocfield.getFieldType()
                                     .equals(entity.getEntityName()))
                                     .filter(assocfield -> assocfield.getRelation() == null).forEach(assocfield -> {
+                                        // skip if the relation is already set for the entity field.
+                                        if (field.getRelation() != null) {
+                                            return;
```

Comment:
This is not required right? since we have checked it in the line 244

Reply:
with the support of adding multiple associations in the same entities. we need to do this.